### DISCUSSION TITLE
[Feature] Send, directly from the history, a mail to ham/spam

### DISCUSSION THREAD

**Discussion by julian70400** [rspamd/rspamd#3567] [General]
Hello,

Would be nice to be able to, directly from the Rspamd webui history, send a mail to the ham/spam scan learning.

Sometime, I see mail subject which are for sure spam content, but which are not sent by the user to the junk or marked as spam. It would be nice to be able to do this, directly from the Rspamd interface.

I am note sure it's possible because not sure if Rspamd have access to the source mail after showing it in the history.

**Comment by adjustGirl** [comment]
That would so fine! +1

**Comment by hsarcon** [comment]
Great feature request +1

**Comment by yardenac** [comment]
I'd find this very convenient too, but wouldn't it mean that rspamd would have to save and archive every email it sees? That seems like not something anybody would really want.

**Comment by Sylvainyo007** [comment]
As rspamd scans each message, it is able to store them for later learning.
For example the ham / spam action from the history could only be possible on messages from the last 24H, or last week, for example.

It could be a very interesting feature !

**Comment by EvDoorne** [comment]
A most wanted, effective and convenient option to have!

**Comment by thompol** [comment]
Issue opened: #4384

**Comment by stonenightcode** [comment]
I also really would appreciate or even donate for this feature! Would be so helpful
